generate_questions.py

```diff
@@ -0,0 +1,74 @@
+import os
+from pathlib import Path
+
+# Function to load LOs and K-levels from a text file
+def load_learning_objectives(file_path):
+    data = []
+    with open(file_path, 'r', encoding='utf-8') as file:
+        for line in file:
+            line = line.strip()
+            if line:
+                # Split by comma and strip any extra spaces from each part
+                lo, k_level = [part.strip() for part in line.split(',')]
+                data.append((lo, k_level))
+    return data
+
+
+# Load the LOs and K-levels from the file 'learning-objectives'
+file_path = 'learning-objectives'  # Provide the correct path to your file
+data = load_learning_objectives(file_path)
+
+# Function to generate the content of the Markdown file
+def generate_md_questions(lo, k_level):
+    # Assign points based on the K-level using a dictionary
+    points_map = {"K1": 1, "K2": 1, "K3": 2, "K4": 3, "K5": 3, "K6": 3}
+    # Default to 0 for any unexpected K-level
+    points = points_map.get(k_level, 0)
+    if points == 0:
+        print(
+            f"Warning: Unexpected K-level '{k_level}' for LO '{lo}'. Defaulting to 0 points.")
+    return f"""# metadata
```

Comment:
Can this be put into a separate variable to enable cleaner code?

Reply:
updated (I hope, that you meant it this way)